### PR TITLE
Add new repo button in repository dropdown

### PR DESCRIPTION
**Pull request by niik**
We've heard in research and feedback that users often look in the repository dropdown for a way to add repositories. This adds an "add" dropdown button to the repository sidebar similar to what we have in the branches dropdown to hopefully alleviate that problem.

screen shot 2018-10-11 at 16 59 58

This addresses one part of #5566 and I plan to tackle the blank slate view when filtering in a follow-up PR.

### CODE REVIEW

**Comment by billygriffin** [comment]
@niik I love that we're doing this! ❤️ 

In looking at the branch dropdown where "New" is next to the input for filtering, we called out that it could be confusing because the two can be conflated. I wonder if similarly here there would be value in separating the filter input from the "Add" button, which might also give a bit more real estate to allow the button to say something more explicit (like "Add repository" instead of "Add"). Thoughts?

**Comment by tierninho** [comment]
The UI was tested for this change, and overall it functions well. 

Two notes:
- Should we arrange the list to mimic the order in the File menu?
- If the user enters text in the filter, should the text be carried through to the repo name (as adding a new branch does this).

Build/OS: 1.4.3-beta0, mac

**Comment by billygriffin** [comment]
> If the user enters text in the filter, should the text be carried through to the repo name (as adding a new branch does this).

My feedback was primarily on this. That seems like it might theoretically be nice, but I'm concerned about the conflation of creating a new repo with filter among existing repos. I think they're separate concerns and while it's a slick implementation to type in a new one, I think separating different actions would help prevent confusion for beginners.

File: app/src/ui/repositories-list/repositories-list.tsx [hunk 1]
```diff
@@ -164,6 +170,54 @@ export class RepositoriesList extends React.Component<
     )
   }
 
+  private renderPostFilter = () => {
+    return (
+      <Button
+        className="new-repository-button"
+        onClick={this.onNewRepositoryButtonClick}
+      >
+        {__DARWIN__ ? 'Add' : 'Add'}
```
Comment (shiftkey) [inline app/src/ui/repositories-list/repositories-list.tsx]: :art: no need for this ternary, right?

**Comment by shiftkey** [comment]
> If the user enters text in the filter, should the text be carried through to the repo name (as adding a new branch does this).

I'd rather keep this change focused on the button and flow, and if we notice a theme of feedback in this vein we can iterate on the UX later.

**Comment by ampinsk** [comment]
> In looking at the branch dropdown where "New" is next to the input for filtering, @ampinsk called out that it could be confusing because the two can be conflated. 

I think in this instance, with the added dropdown, I don't think this is as easily conflated. 

> I wonder if similarly here there would be value in separating the filter input from the "Add" button, which might also give a bit more real estate to allow the button to say something more explicit (like "Add repository" instead of "Add"). Thoughts?

@niik and I chatted through this in Slack, and we had some concerns about exposing all of the actions since it really crowded the menu in a way that made everything harder to parse, and might be confused for a tab menu. These menus weren't really designed to accommodate exposing multiple actions like this so we're in a little bit of a bind until we can maybe redesign these areas more holistically

> That seems like it might theoretically be nice, but I'm concerned about the conflation of creating a new repo with filter among existing repos. I think they're separate concerns and while it's a slick implementation to type in a new one, I think separating different actions would help prevent confusion for beginners.

I don't feel too strongly either way here. I don't think it's necessarily _confusing_, but if we feel strongly that it's conflating concerns I would be fine not having that behavior. Have we heard of anyone being blocked or confused by this behavior?

**Comment by billygriffin** [comment]
> Have we heard of anyone being blocked or confused by this behavior?

It's definitely an assumption I hold because I've always been confused by it. Traditionally an input field next to a button means that the button is a submit for that input field, so that's the source of my confusion. But if y'all disagree with that assumption, I'm fine to continue as is now and test it later.

> @niik and I chatted through this in Slack, and we had some concerns about exposing all of the actions since it really crowded the menu in a way that made everything harder to parse, and might be confused for a tab menu. These menus weren't really designed to accommodate exposing multiple actions like this so we're in a little bit of a bind until we can maybe redesign these areas more holistically

I also don't think we should expose all the options individually, but rather I was suggesting that the verbiage on the dropdown button could be more explicit potentially. So instead of "Add" we might be able to use "Add repository" with the same three options available in the dropdown. Totally fine if we don't think that's better, but just wanted to clarify. 😄

**Comment by shiftkey** [comment]
> So instead of "Add" we might be able to use "Add repository" with the same three options available in the dropdown. Totally fine if we don't think that's better, but just wanted to clarify. 😄

@billygriffin I think we're constrained by the available horizontal space here, so I'm fine with just using "Add" for now until we can think about this area more broadly, like @ampinsk suggested.

**Comment by shiftkey** [comment]
Nothing further raised here, and it's Thursday, so I'm gonna take this in.

**Comment by 101010431** [comment]
/app/styles/ui/_repository-list.scss

**Pull request merged**